### DISCUSSION TITLE
ScriptReference/EditorGUIUtility-textFieldHasSelection

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#244] [Notes]
# ScriptReference/EditorGUIUtility-textFieldHasSelection

https://docs.unity3d.com/ScriptReference/EditorGUIUtility-textFieldHasSelection.html

**Comment by vertxxyz** [comment]
You can use the undocumented [`TextEditor`](https://github.com/Unity-Technologies/UnityCsReference/blob/e740821767d2290238ea7954457333f06e952bad/Modules/IMGUI/TextEditor.cs) class to access all sorts of things relating to text selection in the editor. Including replacing the selection, and the `SelectedText` itself.
```csharp
var editor = (TextEditor)GUIUtility.GetStateObject(typeof(TextEditor), GUIUtility.keyboardControl);
string selectedText = editor.SelectedText;
```
